playbooks/roles/install-oc/tasks/main.yml

```diff
@@ -1,17 +1,4 @@
 ---
 
-- name: Install required packages
-  yum:
-    name:
-      - "bash-completion"
-    state: "present"
-  register: task_result
-  become: true
-
-- name: Install openshift command-line interface
-  shell: "curl -L 'https://mirror.openshift.com/pub/openshift-v3/clients/{{ openshift_version }}.{{ openshift_patch }}/linux/oc.tar.gz' | tar -xvzf - -C /usr/bin oc &>/dev/null"
-  become: true
-
-- name: Configure oc bash command complete
-  shell: "oc completion bash > /etc/profile.d/oc.sh"
-  become: true
+- import_tasks: oc.yml
```

Comment:
rather than checking the variable for every task, decided to move tasks to separate file and imported the file in the main.yml..this is repeated for several roles